libs/langchain/langchain/embeddings/bookend.py

```diff
@@ -0,0 +1,84 @@
+"""Wrapper around Bookend AI embedding models."""
+
+import json
+import requests
+
+from langchain.pydantic_v1 import BaseModel, Field
+from langchain.schema.embeddings import Embeddings
+from typing import Any, List
+
+API_URL = "https://api.bookend.ai/"
+DEFAULT_TASK = "embeddings"
+PATH = "/models/predict"
+
+
+class BookendEmbeddings(BaseModel, Embeddings):
+    """Bookend AI sentence_transformers embedding models.
+
+    Example:
+        .. code-block:: python
+            from langchain.embeddings import BookendEmbeddings
+
+            bookend = BookendEmbeddings(
+                domain={domain}
+                api_token={api_token}
+                model_id={model_id}
+            )
+            bookend.embed_documents([
+                "Please put on these earmuffs because I can't you hear.",
+                "Baby wipes are made of chocolate stardust.",
+            ])
+            bookend.embed_query("She only paints with bold colors; she does not like pastels.")
+    """
+
+    domain: str = Field()
+    """Request for a domain and API token at https://bookend.ai/ to use this embeddings module."""
+    api_token: str = Field()
+    """Request for a domain and API token at https://bookend.ai/ to use this embeddings module."""
+    model_id: str = Field()
+    """Embeddings model ID to use."""
+    auth_header: dict = Field(default_factory=dict)
+
+    def __init__(self, **kwargs: Any):
+        super().__init__(**kwargs)
+        self.auth_header = {"Authorization": "Basic {}".format(self.api_token)}
+
+    def embed_documents(self, texts: List[str]) -> List[List[float]]:
+        """Embed documents using a Bookend deployed embeddings model.
+
+        Args:
+            texts: The list of texts to embed.
+
+        Returns:
+            List of embeddings, one for each text.
+        """
+        result = []
+
+        for text in texts:
+            headers = self.auth_header
+            headers["Content-Type"] = "application/json; charset=utf-8"
```

Comment:
should this just be done once in init?

Reply:
thanks, I'll take it out of the loop and just create/initialize it once